crates/node-db/src/lib.rs

```diff
@@ -345,6 +386,41 @@ pub fn get_validation_progress(conn: &Connection) -> Result<Option<ContentAddres
     Ok(value)
 }
 
+/// Returns the address of block with number 0.
+pub fn get_big_bang_block_address(conn: &Connection) -> Result<Option<ContentAddress>, QueryError> {
+    let mut stmt = conn.prepare(sql::query::GET_BIG_BANG_BLOCK_ADDRESS)?;
+    let value: Option<ContentAddress> = stmt
+        .query_row([], |row| {
+            let block_address: Hash = row.get("block_address")?;
+            Ok(ContentAddress(block_address))
+        })
+        .optional()?;
+    Ok(value)
+}
+
+/// Given a block address, returns the addresses of blocks that have the next block number.
+pub fn get_next_block_address(
+    conn: &Connection,
+    current_block: &ContentAddress,
+) -> Result<Vec<ContentAddress>, QueryError> {
+    let mut stmt = conn.prepare(sql::query::GET_NEXT_BLOCK_ADDRESS)?;
+    let rows = stmt.query_map(
+        named_params! {
+            ":current_block": current_block.0,
+        },
+        |row| {
+            let block_address: Hash = row.get("block_address")?;
+            Ok(block_address)
+        },
+    )?;
+    let mut block_addresses = vec![];
+    for res in rows {
+        let block_address = res?;
+        block_addresses.push(ContentAddress(block_address));
+    }
```

Comment:
Just fyi you can do
```suggestion
    let block_addresses = rows.collect::<Result<Vec<_> , _>>()?;
```

Reply:
Cannot construct `ContentAddress` like that 🤔 Do you recommend doing what you suggested + mapping every hash to `ContentAddress`?

Reply:
oh yeh you would need to map. Up to you though, mostly just a style thing